### ISSUE TITLE
Guidance for customizing equality comparison for code generated with [Record]

### ISSUE THREAD

**Issue by nikolamilekic** [louthy/language-ext#684]
Hi,

I'm having some difficulties with customizing the equality logic that is generated with the `[Record]` attribute. Is it possible to easily customize the logic to perform a structural comparison of any `IEnumerable` fields?

I tried to include a custom `Ord<ImmutableList<MyType>>` in my project (which delegates all calls to `OrdEnumerable`) so that it would be used by `EqDefault`. After some trial and error, I actually got that to work, but unfortunately only while my application is running. The same code running in an NUnit test does not use the custom `Ord`, either because I'm doing something wrong, or because there's a bug in `ClassInstancesAssembly`, or both. From what I can figure, the `ClassInstancesAssembly` does not properly detect my referenced assemblies and so they are not scanned when running from NUnit.

Was I on the right track to begin with? Is there an easier way? The `EqAttribute `in `Attributes.cs` seemed very promising, but I'm either missing something or that attribute is no longer used. 

Any help would be vastly appreciated. Thanks!

**Comment by louthy** [comment]
@nikolamilekic The instance resolution system isn't particularly comprehensive, it will only resolve concrete types right now.

i.e.

```c#
    public struct EqIEnumerableString : Eq<IEnumerable<string>>
    {
        public bool Equals(IEnumerable<string> x, IEnumerable<string> y)
        {
            throw new NotImplementedException();
        }

        public int GetHashCode(IEnumerable<string> x)
        {
            throw new NotImplementedException();
        }
    }
```

So this will be found
```c#
var c1 = Class<Eq<IEnumerable<string>>>.Default;
```

My suggestion if this is really necessary is to use the `Record<TYPE>` system documented on the home-page and override the Equals behaviour manually

**Comment by nikolamilekic** [comment]
Thanks for the suggestion. I chose to go the [Record] route as I found the generated code more intuitive when compared to inheriting from Record<T>. 

I’m not exactly sure what you meant when you talked about concrete types. As I said I got it to work, but only when it’s running in production and not in NUnit. I think that’s due to the way `ClassInstanceAssembly` gets referenced assemblies, but I’m not 100% sure on that. What I know for sure is that my assemblies aren’t even scanned when running from NUnit.
